test/utils/api.spec.ts

```diff
@@ -0,0 +1,72 @@
+import chai from 'chai';
+import sinon from 'sinon';
+
+import API from '../../src/utils/api';
+
+const { expect } = chai;
+const internal = {
+  api: null,
+  options: {
+    getApiKeyAsync: sinon.stub(),
+  },
+  axios: {
+    get: sinon.stub(),
+    defaults: {
+      headers: {
+        common: {},
+      },
+    },
+  },
+};
+
+describe('utils API', () => {
+  beforeEach(async () => {
+    internal.options.getApiKeyAsync.resolves('key');
+    internal.api = new API();
+    await internal.api.init(internal.axios, internal.options);
+  });
+
+  afterEach(() => {
+    internal.axios.get.reset();
+    internal.options.getApiKeyAsync.reset();
+  });
+
+  it('should return null in case of failure', async () => {
+    internal.axios.get.withArgs('/v1/users/current/summaries?start=today&end=today').rejects();
+    const summaries = await internal.api.summaries('today', 'today');
+    expect(summaries).to.equal(null);
+  });
+
+  it('should return null in case of returned status is not 200', async () => {
+    const data = [{ one: 1 }, { two: 2 }];
+    internal.axios.get
+      .withArgs('/v1/users/current/summaries?start=today&end=today')
+      .resolves({ data, status: 400 });
+    const summaries = await internal.api.summaries('today', 'today');
+    expect(summaries).to.equal(null);
+  });
+
+  it('should fetch data from summaries', async () => {
+    const data = [{ one: 1 }, { two: 2 }];
+    internal.axios.get
+      .withArgs('/v1/users/current/summaries?start=today&end=today')
+      .resolves({ data, status: 200 });
+    const summaries = await internal.api.summaries('today', 'today');
+    expect(summaries).to.equal(data);
+  });
+
+  it('should return 0 mins if summaries call fails', async () => {
+    internal.axios.get.withArgs('/v1/users/current/summaries?start=today&end=today').rejects();
+    const todayMins = await internal.api.todayMins('today', 'today');
+    expect(todayMins).to.equal('0 min');
+  });
+
+  it('should fetch data from summaries and return today total time', async () => {
```

Comment:
Can we add one more test to cover for example `2h 30 mins`? (which also includes more than 59 mins).

Reply:
this would be almost copy and paste from this test and just change the value returned from the mock